src/test/java/tests/BaseClass.java

```diff
@@ -0,0 +1,62 @@
+package tests;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class BaseClass {
+    protected WebDriver driver;
+    protected int WAIT_TIMEOUT_SECONDS;
+    Logger logger = Logger.getGlobal();
+
+    protected void waitForPageLoad(WebDriver driver) {
+
+        ExpectedCondition<Boolean> pageLoadCondition = new ExpectedCondition<Boolean>() {
+            public Boolean apply(WebDriver driver) {
+                return ((JavascriptExecutor) driver).executeScript("return document.readyState").equals("complete");
+            }
+        };
+        WebDriverWait wait = new WebDriverWait(driver, WAIT_TIMEOUT_SECONDS);
+        wait.until(pageLoadCondition);
+    }
+
+    @BeforeTest(alwaysRun = true)
+    @Parameters("browser")
+    public void browsersSetup(@Optional("chrome") String browser) throws Exception {
+
+        if (browser.equalsIgnoreCase("firefox")) {
+            System.setProperty("webdriver.gecko.driver", "src\\test\\resources\\webdrivers\\geckodriver.exe");
+            driver = new FirefoxDriver();
+        }
+
+        else if (browser.equalsIgnoreCase("chrome")) {
+            System.setProperty("webdriver.chrome.driver", "src\\test\\resources\\webdrivers\\chromedriver.exe");
+            driver = new ChromeDriver();
+        }
+
+        else if (browser.equalsIgnoreCase("Edge")) {
+            System.setProperty("webdriver.edge.driver", "src\\test\\resources\\webdrivers\\msedgedriver.exe");
+            driver = new EdgeDriver();
+        } else {
+            throw new Exception("Browser is not correct");
+        }
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
```

Comment:
this line is redundant here

Reply:
Refactored, thank you.
Please see in the following commit.